crates/circuit/src/operations.rs

```diff
@@ -269,13 +280,218 @@ impl<'a> Operation for OperationRef<'a> {
     fn directive(&self) -> bool {
         match self {
             Self::Standard(standard) => standard.directive(),
+            Self::StandardInstruction(instruction) => instruction.directive(),
             Self::Gate(gate) => gate.directive(),
             Self::Instruction(instruction) => instruction.directive(),
             Self::Operation(operation) => operation.directive(),
         }
     }
 }
 
+#[derive(Clone, Debug, Copy, Eq, PartialEq, Hash)]
+#[repr(u8)]
+pub enum DelayUnit {
+    NS,
+    PS,
+    US,
+    MS,
+    S,
+    DT,
+}
+
+unsafe impl ::bytemuck::CheckedBitPattern for DelayUnit {
+    type Bits = u8;
+
+    fn is_valid_bit_pattern(bits: &Self::Bits) -> bool {
+        *bits < 6
+    }
+}
+unsafe impl ::bytemuck::NoUninit for DelayUnit {}
+
+impl fmt::Display for DelayUnit {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "{}",
+            match self {
+                DelayUnit::NS => "ns",
+                DelayUnit::PS => "ps",
+                DelayUnit::US => "us",
+                DelayUnit::MS => "ms",
+                DelayUnit::S => "s",
+                DelayUnit::DT => "dt",
+            }
+        )
+    }
+}
+
+impl<'py> FromPyObject<'py> for DelayUnit {
+    fn extract_bound(b: &Bound<'py, PyAny>) -> Result<Self, PyErr> {
+        let str: String = b.extract()?;
+        Ok(match str.as_str() {
+            "ns" => DelayUnit::NS,
+            "ps" => DelayUnit::PS,
+            "us" => DelayUnit::US,
+            "ms" => DelayUnit::MS,
+            "s" => DelayUnit::S,
+            "dt" => DelayUnit::DT,
+            unknown_unit => {
+                return Err(PyValueError::new_err(format!(
+                    "Unit '{}' is invalid.",
+                    unknown_unit
+                )));
+            }
+        })
+    }
+}
+
+/// An internal type used to further discriminate the payload of a `PackedOperation` when its
+/// discriminant is `PackedOperationType::StandardInstruction`.
+///
+/// This is also used to tag standard instructions via the `_standard_instruction_type` class
+/// attribute in the corresponding Python class.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[pyclass(module = "qiskit._accelerate.circuit", eq, eq_int)]
+#[repr(u8)]
+pub(crate) enum StandardInstructionType {
```

Comment:
It is `pub(crate)` here only so that we can expose it to Python from `lib.rs`, IIRC. Perhaps that is not necessary and I was mistaken.

Reply:
If it's needed to export for Python, it's fine (though if it's public to Python, is there no chance that it's useful elsewhere in Rust too?). I mostly just knee-jerk against `pub(crate)`, I think, because it's caused us a fair amount of churn in the last few months.